source/tutorials/packaging-existing-software.md

```diff
@@ -0,0 +1,567 @@
+---
+myst:
+  html_meta:
+    "description lang=en": "Packaging Existing Software With Nix"
+    "keywords": "Nix, packaging"
+---
+
+
+(packaging-existing-software)=
+
+
+# Packaging Existing Software With Nix
+
+One of Nix's primary use-cases is in addressing common difficulties encountered while packaging software, like *managing dependencies*.
+
+In the long term, Nix helps tremendously in alleviating that stress, but when *first* (re)packaging existing software with Nix, it's common to encounter missing dependencies preventing builds from succeeding.
+
+If you haven't already read the tutorial on making a derivation, please go do so before reading this!
+
+In this tutorial, we'll see how to create Nix derivations to package C/C++ software, taking advantage of the [`nixpkgs` `stdenv`](https://nixos.org/manual/nixpkgs/stable/#chap-stdenv) which automates much of the work of building self-contained C/C++ packages.
+
+We'll begin by considering `hello`, a feature-complete implementation of the famous "hello world", which requires no external dependencies.
+
+Then we'll move to progressively more complex packages with their own separate dependencies, leading us to use additional derivation features.
+
+Along the way, we'll encounter and address Nix error messages, build failures, and a host of other issues, developing our iterative debugging techniques as we go.
+
+## Packages in Nix
+Before we proceed, an important point of clarification: we conventionally use the term "package" by analogy to other systems, although this term is not a proper concept in Nix.
+
+For the purposes of this tutorial, by "package" we mean "a Nix function which takes an attribute set of 'dependencies' and produces a derivation", where "dependencies" could be other packages or configuration parameters.
```

Comment:
I prefer to define it in terms of the value returned from the function.
You may well define a package without defining any functions:
```nix
with import <nixpkgs> {};
stdenv.mkDerivation {}
```

Reply:
(but don't do that)

Reply:
> I prefer to define it in terms of the value returned from the function.

I changed it to this; thanks